pytensor/graph/rewriting/utils.py

```diff
@@ -83,8 +84,10 @@ def is_same_graph_with_merge(var1, var2, givens=None):
 
     if givens is None:
         givens = {}
+    givens = dict(givens)
+
     # Copy variables since the MergeOptimizer will modify them.
-    copied = copy.deepcopy([var1, var2, givens])
+    copied = copy.deepcopy((var1, var2, givens))
```

Comment:
Nit: Perhaps inline?

```suggestion
    copied = copy.deepcopy((var1, var2, dict(givens)))
```